src/lib.rs

```diff
@@ -0,0 +1,84 @@
+pub mod read_create {
+    extern crate calamine;
+    use calamine::{open_workbook, Xlsx};
+    use std::fs::File;
+    use std::io::BufReader;
+
+
+pub fn create_file(input_file:&str) -> File{
+
+    let file_handle = File::create(input_file).expect("unable to create a 01-03-2020.txt output file");
+     file_handle
+} 
+
+pub fn open_file(output_file:&str) ->  Xlsx<BufReader<File>>  {
+
+    let workbook: Xlsx<_> = open_workbook(output_file).expect("unable to open the TD_RateCard.xlsx excel file");
+    workbook
+}
+
+}
+
+
+
+pub mod operation{
+    use std::io::Write;
+    use std::fs::File;
+    use std::string::String;
+    extern crate calamine;
+    use calamine::{Reader, Xlsx};
+    use std::io::BufReader;
+
+     pub fn calculation(workbook:&mut Xlsx<BufReader<File>>, out_file:&mut File) {
+    let mut id = 0;
+if let Some(Ok(r)) = workbook.worksheet_range("Sheet1") {
+
+   for column in r.rows() {
+                if id==0 {
+                    id=id+1;
+                    continue;
+                }
+                let rate_index = column[1].to_string();
+                let combo = column[8].to_string();
+                let term_slab = column[6].to_string();
+               let seperated_str: Vec<&str> = combo.split("-").collect();
+              
+               let from_amt = seperated_str[0];
+               let to_amt = seperated_str[1];
+    
+               let mut seperated_space: Vec<&str> = term_slab.split(" ").collect();
+
+               let from_months:i32 = seperated_space[0].parse().unwrap();
+               let from_year = from_months/12;
+               let from_months = from_months % 12;
+               let from_days = seperated_space[2];
+               let to_year;
+               let mut to_months;
+               let to_days;
+               let mut month = String::new();
+               seperated_space.push("mon");
+        if seperated_space[4]!="MONTHS" {
```

Comment:
to check something always use to_uppercase() or to_lowercase() like `if seperated_space[4].to_uppercase() != "MONTHS"`